ios/Tests/NetworkTests.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  NetworkTests.swift
+//  YourFirstCommitTests
+//
+//  Created by Alex Antipov on 08.02.2023.
+//
+
+import XCTest
+@testable import YourFirstCommit
+
+final class NetworkTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+    
+    func testFirstCommitLoading() async throws {
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct results.
+        // Any test you write for XCTest can be annotated as throws and async.
+        // Mark your test throws to produce an unexpected failure when your test encounters an uncaught error.
+        // Mark your test async to allow awaiting for asynchronous code to complete. Check the results with assertions afterwards.
+        
+        let repo: Repo = Repo(id: 1, name: "pomodoress", fullName: "r-ss/pomodoress", htmlUrl: "h-ti-ti")
+        
+        
+        Task(priority: .background) {
+            let result = await GithubService().loadFirstCommit(for: repo)
```

Comment:
The point of writing unit tests is to verify the business logic flow a part of your code has passed. 

In this scenario, the `GithubService` uses `HTTPClient` as a client to make any network request. `HTTPClient` uses the real `URLSession.shared` to request the data from the passed endpoint. That's a bad practice to reach out to the real server. Instead, you need to mock up any part of the code that interacts with real-world things, e.g. network requests, I/O operation, etc. 

I see 2 options there. First, you can use the `GithubServicable` protocol and create a standalone mock class to handle a kind of response represented as local `.json` file. Otherwise, you can mock up URLSession within a new protocol and control the output of the `URLSession.shared.data(for: request, delegate: nil)` method. Anyway you need to hide the real-world network implementation out the test case.